frame/curve-amm/src/lib.rs

```diff
@@ -824,6 +823,21 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Return pool information for given pool_id.
+		pub fn get_pool_info(pool_id: PoolId) -> Option<StableSwapPoolInfo<T::AccountId>> {
+			let pool_data = Pools::<T>::get(pool_id)?;
+			let pool;
+			match pool_data {
+				PoolData::StableSwapPoolData(pool_info) => {
+					pool = pool_info;
+				},
+				_ => {
+					panic!("Only stable swap pool expected.");
```

Comment:
could we avoid this panic here? it feels not good for stable to be aware of other curves?

Reply:
should it generate Error?

Reply:
we MUST NEVER panic